lib/queue.js

```diff
@@ -851,49 +814,38 @@ Queue.prototype.run = function(concurrency) {
   This function updates the delay timer, which is a timer that timeouts
   at the next known delayed job.
 */
-Queue.prototype.updateDelayTimer = function(newDelayedTimestamp) {
-  const now = Date.now();
-
-  newDelayedTimestamp = Math.round(newDelayedTimestamp);
-  if (newDelayedTimestamp < this.delayedTimestamp) {
-    clearTimeout(this.delayTimer);
-    this.delayedTimestamp = newDelayedTimestamp;
+Queue.prototype.updateDelayTimer = function() {
+  return scripts
+    .updateDelaySet(this, Date.now())
+    .then(nextTimestamp => {
+      this.delayedTimestamp = nextTimestamp
+        ? nextTimestamp / 4096
+        : Number.MAX_VALUE;
+
+      // Clear any existing update delay timer
+      if (this.delayTimer) {
+        clearTimeout(this.delayTimer);
+      }
 
-    const nextDelayedJob = newDelayedTimestamp - now;
-    const delay = nextDelayedJob <= 0 ? 0 : nextDelayedJob;
+      // Delay for the next update of delay set
+      const delay = _.min([
+        this.delayedTimestamp - Date.now(),
+        this.settings.guardInterval
+      ]);
 
-    const update = delay => {
-      if (delay) {
-        delay = delay < MAX_TIMEOUT_MS ? delay : MAX_TIMEOUT_MS;
-        this.delayTimer = setTimeout(delayUpdate, delay);
+      // Schedule next processing of the delayed jobs
+      if (delay <= 0) {
+        // Next set of jobs are due right now, process them also
+        this.updateDelayTimer();
```

Comment:
I'm getting warnings about unhandled promises stemming from this. It seems like this should return here?

Reply:
Nevermind, I see `updateDelayTimer` handles errors itself. I must be running into something else.